File: benchmarks/NSubstitute.Analyzers.Benchmarks.CSharp.Source/DiagnosticsSources/NonSubstitutableMemberReceivedDiagnosticsSource.cs
using NSubstitute.Analyzers.Benchmarks.CSharp.Source.Models;

namespace NSubstitute.Analyzers.Benchmarks.CSharp.Source.DiagnosticsSources
{
    public class NonSubstitutableMemberReceivedDiagnosticsSource
    {
        public void NS1001_NonVirtualSetupSpecification()
        {
            var substitute = Substitute.For<Foo>();

            substitute.Received(1).ObjectReturningMethod();
            _ = substitute.Received(1).Property;
            _ = substitute.Received(1)[0];

            SubstituteExtensions.Received(substitute, 1).ObjectReturningMethod();
            _ = SubstituteExtensions.Received(substitute, 1).Property;
            _ = SubstituteExtensions.Received(substitute, 1)[0];
        }

        public void NS1003_InternalVirtualSetupSpecification()
        {
            var substitute = Substitute.For<Foo>();

            substitute.Received(1).InternalObjectReturningMethod();
            _ = substitute.Received(1).InternalObjectReturningProperty;
            _ = substitute.Received(1)[0, 0, 0];

            SubstituteExtensions.Received(substitute, 1).InternalObjectReturningMethod();
            _ = SubstituteExtensions.Received(substitute, 1).InternalObjectReturningProperty;
            _ = SubstituteExtensions.Received(substitute, 1)[0, 0, 0];
        }
    }
}

File: benchmarks/NSubstitute.Analyzers.Benchmarks.VisualBasic.Source/DiagnosticsSources/NonSubstitutableMemberReceivedDiagnosticsSource.vb
Imports NSubstitute.Analyzers.Benchmarks.Shared.Models

Namespace DiagnosticsSources
    Public Class NonSubstitutableMemberReceivedDiagnosticsSource
        Public Sub NS1001_NonVirtualSetupSpecification()
            Dim substitute = NSubstitute.Substitute.[For](Of Foo)()
            substitute.Received(1).ObjectReturningMethod()
            Dim a = substitute.Received(1).[Property]
            Dim b = substitute.Received(1)(0)
            SubstituteExtensions.Received(substitute, 1).ObjectReturningMethod()
            Dim c  = SubstituteExtensions.Received(substitute, 1).[Property]
            Dim d = SubstituteExtensions.Received(substitute, 1)(0)
        End Sub

        Public Sub NS1003_InternalVirtualSetupSpecification()
            Dim substitute = NSubstitute.Substitute.[For](Of Foo)()
            substitute.Received(1).InternalObjectReturningMethod()
            Dim a = substitute.Received(1).InternalObjectReturningProperty
            Dim b = substitute.Received(1)(0, 0, 0)
            SubstituteExtensions.Received(substitute, 1).InternalObjectReturningMethod()
            Dim c = SubstituteExtensions.Received(substitute, 1).InternalObjectReturningProperty
            Dim d = SubstituteExtensions.Received(substitute, 1)(0, 0, 0)
        End Sub
    End Class
End Namespace
